pkg/db/db.go

```diff
@@ -0,0 +1,413 @@
+			}
+			return column, nil
+		}
+	}
+
+	// no unique index found, seek index with max cardinality
+	var c *model.ColumnInfo
+	var maxCardinality int
+	for _, fields := range rowsData {
+		if string(fields["Seq_in_index"]) == "1" {
+			cardinality, err := strconv.Atoi(string(fields["Cardinality"]))
+			if err != nil {
+				return nil, errors.Trace(err)
+			}
+			if cardinality > maxCardinality {
+				column, valid := GetColumnByName(tableInfo, string(fields["Column_name"]))
+				if !valid {
+					return nil, errors.New(fmt.Sprintf("can't find column %s in %s.%s", string(fields["Column_name"]), dbName, table))
+				}
+				maxCardinality = cardinality
+				c = column
+			}
+		}
+	}
+
+	return c, nil
+}
+
+// GetOrderKey return some columns for order
+func GetOrderKey(tbInfo *model.TableInfo) ([]string, []*model.ColumnInfo) {
+	keys := make([]string, 0, 2)
+	keyCols := make([]*model.ColumnInfo, 0, 2)
+
+	for _, index := range tbInfo.Indices {
+		if index.Primary {
+			for _, indexCol := range index.Columns {
+				keys = append(keys, indexCol.Name.O)
+			}
+		}
+	}
+
+	if len(keys) == 0 {
+		// no primary key found, use all fields as order by key
+		for _, col := range tbInfo.Columns {
+			keys = append(keys, col.Name.O)
+			keyCols = append(keyCols, col)
+		}
+	} else {
+		for _, col := range tbInfo.Columns {
+			for _, key := range keys {
+				if col.Name.O == key {
+					keyCols = append(keyCols, col)
+				}
+			}
+		}
+	}
+	return keys, keyCols
+}
+
+// GetFirstColumn returns the first column in the table
+func GetFirstColumn(db *sql.DB, dbname string, table string) (*model.ColumnInfo, error) {
+	tableInfo, err := GetSchemaTable(db, dbname, table)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	return tableInfo.Columns[0], nil
+}
+
+// GetRandomValues returns some random value of a table
+func GetRandomValues(db *sql.DB, dbname string, table string, field string, num int64, min, max interface{}, timeRange string) ([]string, error) {
+	randomValue := make([]string, 0, num)
+	query := fmt.Sprintf("SELECT `%s` FROM (SELECT `%s` FROM `%s`.`%s` WHERE `%s` > \"%v\" AND `%s` < \"%v\" AND %s ORDER BY RAND() LIMIT %d)rand_tmp ORDER BY `%s`",
+		field, field, dbname, table, field, min, field, max, timeRange, num, field)
+	log.Infof("get random values sql: %s", query)
+	rows, err := db.Query(query)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		var value string
+		err = rows.Scan(&value)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		randomValue = append(randomValue, value)
+	}
+
+	return randomValue, nil
+}
+
+func GetColumnByName(table *model.TableInfo, name string) (*model.ColumnInfo, bool) {
+	var c *model.ColumnInfo
+	for _, column := range table.Columns {
+		if column.Name.O == name {
+			c = column
+			break
+		}
+	}
+
+	if c != nil {
+		return c, true
+	}
+
+	return nil, false
```

Comment:
It seems we check existence with `c ==nil`, so no need for additional boolean value. It's OK to just return `*model.ColumnInfo`.